operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +60,21 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+// ManagedBootImageConfig contains configuration data for the bootimages management feature
+// +union
+type ManagedBootImageConfig struct {
+	// mode has three settings:
+	// enabled - MCO will manage all machinesets
+	// matchSelector - MCO will manage machinesets that match with the label selector
+	// disabled - MCO will not manage any machinesets
+	// If unset, mode will default to disabled
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Enabled";"Disabled";"MatchSelector"
+	// +kubebuilder:validation:Required
+	Mode string `json:"mode,omitempty"`
+
+	// matchSelector specifies a label selector for MachinesSets that will be managed by this mechanism
+	// +optional
+	MatchSelector *metav1.LabelSelector `json:"matchSelector,omitempty"`
```

Comment:
Are we sure this is the member we want, rather than creating a struct? With this, we have absolutely no way to evolve the API down the line, we cannot add new ways of matching a selector, we would have to add a new `mode`. That may be ok but something to consider, do we expect that MatchSelector may ever need more fields than what exists here today?

Reply:
I thought a little bit about this, I can't think of any future customization as of now. But I would like to leave room for something we haven't planned for - so I made it a struct and enclosed the selector within that as you advised.

Also did some renaming so it makes sense with the new struct. If I'm overlooking some existing naming conventions, please let me know...I'm not a 100% satisfied with `ManagedBootImageConfigCustomization` ... it is quite a mouthful :(